### PR TITLE
Replace old network exception with WL exceptions, populate nonce for Id Token flow.

### PR DESCRIPTION
**Pull request by nithyaganeshng**
**Problem:**

- Nonce which is required for Id Token flow wasn't populated internally. 
- Conversion of exception between legacy code and WL was failing.

**Solution:**

- Populated the nonce in Id Token requirement so that it can be used to complete the flow.
- Added a check for newer exceptions while handling exception conversion between native and custom result object.

**Validation:**
Validated using unit tests and manually.

**Type of change:**
- [ ] Feature work
- [x] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [ ] Test
- [ ] Logging/Telemetry

**Risk**:
- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [x] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [ ] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

**Work Item links**:
Please include here links for this work item, or deferred work, or related work. E.g. if the refactoring is too big to fit in this PR, or the localized strings need to be updated later, please link the TODO work items here.

**Documentation Links**:
Please include here links to any related background documentation for this PR.

### CODE REVIEW

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/util/controlflow/Result.kt [hunk 1]
```diff
@@ -23,10 +25,16 @@ internal fun <T> KotlinResult<T>.toSDK(): Result<T> {
...
-            return Result.Failure(SdkException("Could not cast failure to SDK", it.cause))
+        when (it) {
+            is SdkException -> {
+                return Result.Failure(it)
+            }
+            is NetworkingException -> {
+                return Result.Failure(SdkException("Networking Exception", it))
+            }
+            else -> {
+                return Result.Failure(SdkException("Could not cast failure to SDK", it.cause))
```
Comment (symorton) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/did/sdk/util/controlflow/Result.kt]: is It possible to change the string "Could not cast failure to SDK" to be like the message that describes the failure or something?

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/IdTokenRequirement.kt [hunk 2]
```diff
@@ -24,7 +24,7 @@ class IdTokenRequirement(
     val scope: String,
 
     // Nonce is generated using user DID.
-    val nonce: String,
+    private var _nonce: String,
```
Comment (symorton) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/IdTokenRequirement.kt:27]: why prefix nonce with _? Is this a kotlin convention?

Comment (nithyaganeshng) [reply walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/IdTokenRequirement.kt:27]: yes, it is to represent a backing field, and the public field has the normal naming.

Comment (logangirvin) [reply walletlibrary/src/main/java/com/microsoft/walletlibrary/requests/requirements/IdTokenRequirement.kt:27]: There are two "nonce" variables. the private "_nonce" holds the value, while the getter for nonce and theoretically a setter but in this case a function can do additional checks.

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/util/http/httpagent/IResponse.kt [hunk 3]
```diff
@@ -57,7 +58,7 @@ class IResponse(
         }
         return NetworkingException(
             message = message,
-            code = code,
+            code = VerifiedIdExceptions.NETWORKING_EXCEPTION.value,
             correlationId = correlationId,
             statusCode = status.toString(),
```
Comment (logangirvin) [inline walletlibrary/src/main/java/com/microsoft/walletlibrary/util/http/httpagent/IResponse.kt:63]: This may need to be `code.toString()`

**Review by logangirvin — DISMISSED** [review]

**Review by logangirvin — APPROVED** [review]

File: walletlibrary/src/main/java/com/microsoft/walletlibrary/util/http/httpagent/IResponse.kt (hunk 3 above)
Comment (nithyaganeshng) [reply walletlibrary/src/main/java/com/microsoft/walletlibrary/util/http/httpagent/IResponse.kt:63]: We have two different codes for Networking exception. statusCode is the http error code. code is from the enum of error codes for WL.

**Pull request merged**